docs/src/content/docs/reference/overrides.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Overrides Reference
+description: Learn
+---
+
+You can override Starlight’s built-in components by providing paths to replacement components in Starlight’s [`components`](/reference/configuration#components) configuration option.
+This page lists all components available to override and links to their default implementations on GitHub.
+
+Learn more in the [Guide to Overriding Components](/guides/overriding-components/).
+
+## Prop types
+
+All components can access a standard `Astro.props` object that contains information about the current page.
+
+To type your custom components, import the `Props` type from Starlight:
+
+```astro
+---
+import type { Props } from '@astrojs/starlight/props';
+
+const { hasSidebar } = Astro.props;
+//      ^ type: boolean
+---
+```
+
+This will give you autocomplete and types when accessing `Astro.props`.
+
+### `Props`
+
+```ts
+interface Props {
+  /** Page writing direction. */
+  dir: 'ltr' | 'rtl';
+
+  /** BCP-47 language tag for this page’s locale. */
+  lang: string;
+
+  /** The base path at which a language is served. `undefined` for root locale slugs. */
```

Comment:
even this line requires horizontal scroll, but the whole code sample is too tall to display the horizontal scroll bar at this point. So, I think this will all have to be text, not code. OR, everything longer than this line for sure needs to be split across two lines (and even then, I'm not sure if that works for everyone's device)